### PR TITLE
Mcga 10

### PR DESCRIPTION
**Pull request by Taimoorrana1**

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -38,6 +38,9 @@ android {
 dependencies {
 
     compile fileTree(dir: 'libs', include: ['*.jar'])
+
+
+
```
Comment (arekmano) [inline app/build.gradle]: why?

File: app/src/main/java/com/concordia/mcga/fragments/NavigationFragment.java [hunk 2]
```diff
@@ -39,6 +57,22 @@
 
     //Outdoor Map
     private final float CAMPUS_DEFAULT_ZOOM_LEVEL = 16f;
+    LocationListener gpsListen = new LocationListener() {
+        public void onLocationChanged(Location location) {
+            //Method called when new location is found by the network
+
+            Log.d("Message: ", "Location changed," + location.getLatitude() + "," + location.getLongitude() + ".");
```
Comment (arekmano) [inline app/src/main/java/com/concordia/mcga/fragments/NavigationFragment.java]: We only log the location?

File: app/src/main/java/com/concordia/mcga/fragments/NavigationFragment.java [hunk 3]
```diff
@@ -68,8 +107,42 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
...
-
         //Init View Components
+        navigationSearch = (EditText) parentLayout.findViewById(R.id.navigationSearch);
+        mapCenterButton = (FloatingActionButton) parentLayout.findViewById(R.id.mapCenterButton);
+        mapCenterButton.setOnClickListener(new OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                Log.d("Testing mapCenterButton", "Initializing OnClickListener");
+                if (!gpsmanager.isProviderEnabled(LocationManager.GPS_PROVIDER)) {
+                    Log.d("Testing AlertGPS Launch", "Initializing method");
```
Comment (arekmano) [inline app/src/main/java/com/concordia/mcga/fragments/NavigationFragment.java]: too many debugs :| Are they all needed?

File: app/src/main/java/com/concordia/mcga/fragments/NavigationFragment.java [hunk 4]
```diff
@@ -128,6 +201,8 @@ public void onMapReady(GoogleMap googleMap) {
 
         //Settings
         map.getUiSettings().setMapToolbarEnabled(false);
+        map.getUiSettings().setAllGesturesEnabled(true);
```
Comment (arekmano) [inline app/src/main/java/com/concordia/mcga/fragments/NavigationFragment.java]: why is this necessary?

File: app/src/test/java/com/concordia/mcga/fragments/NavigationFragmentTest.java [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+@RunWith(RobolectricTestRunner.class)
+@Config(constants = BuildConfig.class, shadows = ShadowBitmapDescriptorFactory.class)
+public class NavigationFragmentTest {
+    private NavigationFragment fragment;
+
+    @Before
+    public void setUp() throws Exception {
+        LayoutInflater fakeInflater = Mockito.mock(LayoutInflater.class);
+        ViewGroup fakeContainer =  Mockito.mock(ViewGroup.class);
+        //View fakeToolbar = Mockito.mock(View.class);
```
Comment (arekmano) [inline app/src/test/java/com/concordia/mcga/fragments/NavigationFragmentTest.java]: dead code

File: app/src/test/java/com/concordia/mcga/fragments/NavigationFragmentTest.java [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+        //Bundle fakeBundle = Mockito.mock(Bundle.class);
+        //AppCompatImageButton fakeCancelButton = Mockito.mock(AppCompatImageButton.class);
+        IndoorMapFragment fakeIndoors = Mockito.mock(IndoorMapFragment.class);
+
+        Button fakeButton = Mockito.mock(Button.class);
+
+        when(fakeLayout.findViewById(R.id.viewSwitchButton)).thenReturn(fakeButton);
+        //when(fakeLayout.findViewById(R.id.nav_toolbar)).thenReturn(fakeToolbar);
+        //when(fakeToolbar.findViewById(R.id.search_location_button)).thenReturn(fakeCancelButton);
+        //when(fakeToolbar.findViewById(R.id.search_destination_button)).thenReturn(fakeCancelButton);
```
Comment (arekmano) [inline app/src/test/java/com/concordia/mcga/fragments/NavigationFragmentTest.java]: dead

File: app/src/test/java/com/concordia/mcga/fragments/NavigationFragmentTest.java [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+        //viewSwitchButton = (Button) parentLayout.findViewById(R.id.viewSwitchButton);
+        //SupportFragmentTestUtil.startFragment(fragment);
+
+    }
+
+    @Test
+    public void shouldNotBeNull() throws Exception
+    {
+        assertNotNull(fragment);
+        assertNotNull(fragment.getView());
```
Comment (arekmano) [inline app/src/test/java/com/concordia/mcga/fragments/NavigationFragmentTest.java]: more tests please

File: app/build.gradle [hunk 8]
```diff
@@ -51,6 +54,7 @@ dependencies {
     testCompile group: 'org.khronos', name: 'opengl-api', version: 'gl1.1-android-2.1_r1'
 
     compile 'com.android.support:design:24.1.1'
+    compile 'com.android.support:design:24.0.0'
```
Comment (Taimoorrana1) [inline app/build.gradle]: why do we need 2 different versions?

File: app/src/main/java/com/concordia/mcga/fragments/NavigationFragment.java [hunk 9]
```diff
@@ -286,6 +359,54 @@ public void notifyObservers() {
         }
     }
 
+    public static boolean AlertGPS(final Activity activity) { //GPS detection method
```
Comment (Taimoorrana1) [inline app/src/main/java/com/concordia/mcga/fragments/NavigationFragment.java]: Javadoc.
Method name should start with lowercase

File: app/src/main/java/com/concordia/mcga/fragments/NavigationFragment.java [hunk 10]
```diff
@@ -286,6 +359,54 @@ public void notifyObservers() {
...
+                    public void onClick(DialogInterface dialog, int id) {
+                        dialog.cancel();
+                    }
+                });
+        AlertDialog alert = build.create();
+        alert.show();
+        return true;
+    }
+
+    public static boolean locateMe(GoogleMap map, Activity activity, LocationManager gpsmanager, LocationListener gpsListen) {
```
Comment (Taimoorrana1) [inline app/src/main/java/com/concordia/mcga/fragments/NavigationFragment.java]: javadoc

File: app/build.gradle (hunk 1 above)
Comment (mihaiqc) [inline app/build.gradle]: Remnant of a dependency I previously added to attempt to test Navigation Fragment. I left the white space mistakenly.

File: app/src/main/java/com/concordia/mcga/fragments/NavigationFragment.java (hunk 3 above)
Comment (mihaiqc) [inline app/src/main/java/com/concordia/mcga/fragments/NavigationFragment.java]: These are mostly log remnants from when I was testing an enormous amount of methods. The pull request was done before I could clear them. Most of them are unnecessary and will be removed.

File: app/src/main/java/com/concordia/mcga/fragments/NavigationFragment.java (hunk 2 above)
Comment (mihaiqc) [inline app/src/main/java/com/concordia/mcga/fragments/NavigationFragment.java]: This is an abstract void used to update the location.

File: app/src/test/java/com/concordia/mcga/fragments/NavigationFragmentTest.java (hunk 7 above)
Comment (mihaiqc) [inline app/src/test/java/com/concordia/mcga/fragments/NavigationFragmentTest.java]: Yes, as discussed during the meeting, I'm on it.

File: app/src/main/java/com/concordia/mcga/fragments/NavigationFragment.java (hunk 9 above)
Comment (mihaiqc) [inline app/src/main/java/com/concordia/mcga/fragments/NavigationFragment.java]: Updated

File: app/src/main/java/com/concordia/mcga/fragments/NavigationFragment.java (hunk 10 above)
Comment (mihaiqc) [inline app/src/main/java/com/concordia/mcga/fragments/NavigationFragment.java]: Updated
